### PR TITLE
refactor: implement `fromEmitterEvent`

### PR DESCRIPTION
**Pull request by jrandolf-google**

### CODE REVIEW

File: packages/puppeteer-core/src/api/Page.ts [hunk 1]
```diff
@@ -1677,15 +1677,15 @@ export abstract class Page extends EventEmitter<PageEvents> {
     requestsInFlight = 0
   ): Observable<void> {
     return merge(
-      fromEvent(
+      fromEmitterEvent(
         networkManager,
         NetworkManagerEvent.Request as unknown as string
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/api/Page.ts]: Can we remove the type casting now?

File: packages/puppeteer-core/src/common/util.ts [hunk 2]
```diff
@@ -594,9 +593,7 @@ export async function waitForHTTP<T extends {url(): string}>(
   cancelation: Deferred<never>
 ): Promise<T> {
   return await firstValueFrom(
-    (
-      fromEvent(networkManager, eventName as unknown as string) as Observable<T>
-    ).pipe(
+    (fromEmitterEvent(networkManager, eventName) as Observable<T>).pipe(
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/util.ts:596]: Can we use `eventName: NetworkManagerEvents`? and remove this type casting?

Comment (jrandolf) [reply packages/puppeteer-core/src/common/util.ts:596]: Looks non-trivial.

**Review by Lightning00Blade — APPROVED** [review]

File: packages/puppeteer-core/src/common/util.ts [hunk 3]
```diff
@@ -711,3 +708,18 @@ function convertPrintParameterToInches(
   }
   return pixels / unitToPixels[lengthUnit];
 }
+
+export function fromEmitterEvent<
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/util.ts:715]: Please create a TSDoc with internal

File: packages/puppeteer-core/src/common/util.ts [hunk 4]
```diff
@@ -9,13 +9,12 @@ import type {Readable} from 'stream';
 
 import type {Protocol} from 'devtools-protocol';
 
-import type {Observable} from '../../third_party/rxjs/rxjs.js';
+import {Observable} from '../../third_party/rxjs/rxjs.js';
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/util.ts]: Missed this can you move it to a single import pls?

**Pull request merged**